scripts/analyze.py

```
@@ -0,0 +1,717 @@
+    """Run CodeChecker analysis."""
+    # Check all dependencies upfront
+    check_codechecker_dependencies()
+
+    preset = args.preset
+    jobs = args.jobs or os.cpu_count() or 4
+    compile_commands = require_compile_commands(preset)
+
+    print("Running CodeChecker comprehensive analysis:")
+    print(
+        f"  - clang-tidy: {len(CRITICAL_TIDY_CHECKS)} critical + {len(ADDITIONAL_TIDY_CHECKS)} additional checks")
+    print(f"  - clangsa: All checks including experimental (deep mode + CTU)")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)